File: react-native-mobilesdk/android/src/main/java/com/mobilesdk/survey/MobileSdkSurveyManager.kt
package com.mobilesdk.survey

class MobileSdkSurveyManager {
    companion object {
        private val activeSurvey: MutableMap<String, MobileSdkSurveyWrapper> = mutableMapOf()
    }

    fun getSurvey(serverId: String, programKey: String, surveyId: String): MobileSdkSurveyWrapper? {
        return activeSurvey[getKey(serverId, programKey, surveyId)]
    }

    fun addSurvey(serverId: String, programKey: String, surveyId: String, survey: MobileSdkSurveyWrapper) {
        activeSurvey[getKey(serverId, programKey, surveyId)] = survey
    }

    fun removeSurvey(serverId: String, programKey: String, surveyId: String) {
        activeSurvey.remove(getKey(serverId, programKey, surveyId))
    }

    private fun getKey(serverId: String, programKey: String, surveyId: String): String {
        return "${serverId}_${programKey}_$surveyId"
    }
}

File: react-native-mobilesdk/ios/Survey/MobileSdkSurveyManager.swift
import Foundation

class MobileSdkSurveyManager {
  private static var activeSurvey: [String: MobileSdkSurveyWrapper] = [:]
  
  func getSurvey(serverId: String, programKey: String, surveyId: String) -> MobileSdkSurveyWrapper? {
    return MobileSdkSurveyManager.activeSurvey[getKey(serverId: serverId, programKey: programKey, surveyId: surveyId)]
  }
  
  func addSurvey(serverId: String, programKey: String, surveyId: String, survey: MobileSdkSurveyWrapper) {
    MobileSdkSurveyManager.activeSurvey[getKey(serverId: serverId, programKey: programKey, surveyId: surveyId)] = survey
  }
  
  func removeSurvey(serverId: String, programKey: String, surveyId: String) {
    MobileSdkSurveyManager.activeSurvey.removeValue(forKey: getKey(serverId: serverId, programKey: programKey, surveyId: surveyId))
  }
  
  private func getKey(serverId: String, programKey: String, surveyId: String) -> String {
    return "\(serverId)_\(programKey)_\(surveyId)"
  }
}
